pkg/fanal/analyzer/sbom/sbom.go

```diff
@@ -39,24 +40,32 @@ func (a sbomAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (
 		return nil, xerrors.Errorf("SBOM decode error: %w", err)
 	}
 
-	// For Bitnami images
+	// Bitnami images
+	// SPDX files are located under the /opt/bitnami/<component> directory
+	// and named with the pattern .spdx-<component>.spdx
+	// ref: https://github.com/bitnami/vulndb#how-to-consume-this-cve-feed
 	if strings.HasPrefix(input.FilePath, "opt/bitnami/") {
-		dir, file := path.Split(input.FilePath)
-		bin := strings.TrimPrefix(file, ".spdx-")
-		bin = strings.TrimSuffix(bin, ".spdx")
-		binPath := path.Join(input.FilePath, "../bin", bin)
+		componentPath := path.Dir(input.FilePath)
 		for i, app := range bom.Applications {
-			// Replace the SBOM path with the binary path
-			bom.Applications[i].FilePath = binPath
+			switch app.Type {
+			case ftypes.Bitnami:
+				// Replace the SBOM path with the component path
+				bom.Applications[i].FilePath = componentPath
+			default:
+				bom.Applications[i].Type = app.Type
+			}
 
 			for j, pkg := range app.Libraries {
+				// package file path might be empty if "filesAnalyzed" is false
```

Comment:
We need file paths. Otherwise, users cannot understand where the package comes from. I remember the Bitnami SBOM used to include file paths. @juan131 Any reasons for removing file paths from SBOM?

Reply:
Apparently this is sth that was already discussed at https://github.com/aquasecurity/trivy/discussions/4507

Reply:
Yes, after the fix we continued facing issues with the validator. It complained about `filesAnalyzed` being set to `true` while missing the package verification code:
```
The following warning(s) were raised: [Relationship error: Relationship error: Relationship error: Missing required package verification code for package co.elastic.apm:apm-agent in jar in jar in Elasticsearch in Elasticsearch in SPDX document for Elasticsearch 8.9.1]
```
See: https://spdx.github.io/spdx-spec/v2.3/package-information/#79-package-verification-code-field
And.. if we force `filesAnalyzed` to `false` we find the error below:
```
The following warning(s) were raised: [Relationship error: Relationship error: Relationship error: Warning: Found analyzed files for package co.elastic.apm:apm-agent when analyzedFiles is set to false. in jar in jar in Elasticsearch in Elasticsearch in SPDX document for Elasticsearch 8.9.1]
```
Spec: https://spdx.github.io/spdx-spec/v2.3/package-information/#78-files-analyzed-field

Reply:
Thanks for the information. We'll look into it.

Reply:
@DmitriyLewen Can you please take a look? If we confirm the problem, we should create an issue and fix it.

Reply:
I checked our implementation:
we don't use `filesAnalyzed` (always `false`) and `packageVerificationCode` fields.
We add files to [Files](https://spdx.github.io/spdx-spec/v2.3/file-information/) and use them in [relationships](https://spdx.github.io/spdx-spec/v2.3/relationships-between-SPDX-elements/).

But perhaps it looks more correct to use `fileAnalyzed`. Perhaps we need to consider using this field.

Reply:
>Yes, after the fix we continued facing issues with the validator.

@juan131 Our SPDX is valid. I guess you build SBOM on top of our SBOM, right?

Reply:
> Our SPDX is valid

We obtain warnings in the [SPDX validator](https://tools.spdx.org/app/validate/) such as the ones below with your SPDX:

```
Missing required package verification code for package XXX in YYY
(...)
Invalid license id 'Artistic'. Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'
```

> I guess you build SBOM on top of our SBOM, right?

Yes, we use your SBOM to build our SPDX files.